src/Employer/Employer.Web/Services/LevyDeclarationCookieWriter.cs

```diff
@@ -0,0 +1,41 @@
+using Esfa.Recruit.Employer.Web.Configuration;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+
+namespace Esfa.Recruit.Employer.Web.Services
+{
+    public class LevyDeclarationCookieWriter
+    {
+        private readonly IHostingEnvironment _hostingEnvironment;
+        private readonly IDataProtector _dataProtector;
+
+        public LevyDeclarationCookieWriter(IHostingEnvironment hostingEnvironment, IDataProtectionProvider dataProtectionProvider)
+        {
+            _hostingEnvironment = hostingEnvironment;
+            _dataProtector = dataProtectionProvider.CreateProtector(DataProtectionPurposes.LevyDeclarationCookie);
+        }
+
+        public string GetCookieFromRequest(HttpRequest request)
+        {
+            var content = request.Cookies[CookieNames.LevyEmployerIndicator];
+
+            if (string.IsNullOrWhiteSpace(content))
+                return null;
+
+            return _dataProtector.Unprotect(content);
+        }
+
+        public void WriteCookie(HttpResponse response, string userId, string employerAccountId)
+        {
+            var protectedValue = _dataProtector.Protect($"{userId}-{employerAccountId}");
+
+            response.Cookies.Append(CookieNames.LevyEmployerIndicator, protectedValue, EsfaCookieOptions.GetDefaultHttpCookieOption(_hostingEnvironment));
```

Comment:
Can you make sure we delete this cookie on signout if if it to be repopulated on sign in.

Reply:
I think we need to manage our cookies better. Not 💯 on this one being an issue. But any cookie that we do not define an expiry for, we should make sure we clean up. Some of the old dashboard filter cookies are hanging around on user machines and we want to remove the listing from the MA page.

Reply:
You're right. But in this scenario there's a benefit of not deleting/expiring. The functionality for selecting an option means that the cookie is always valid. i.e. they aren't able to go from selecting 'yes' to 'no'.